components/num-util/src/fixed_decimal.rs

```diff
@@ -406,8 +405,8 @@ impl FromStr for FixedDecimal {
         // characters are digits and if you have at most one dot
         // Note: Input of format 111_123 is detected as syntax error here
         // Note: Input starting or ending with a dot is detected as syntax error here (Ex: .123, 123.)
-        for (i, c) in no_sign_str.bytes().enumerate() {
-            if c == b'.' {
+        for (i, c) in no_sign_str.iter().enumerate() {
+            if *c == b'.' {
```

Comment:
doesn't it copy here? Wouldn't it be faster to do `&b'.'`? I'm not sure tbh, just wondering.

Reply:
My 2 cents that this iteration is all done in registers.